File: rust/src/classes/archetype_rank_abilities/standard_modifiers.rs
use crate::classes::archetype_rank_abilities::RankAbility;



pub fn add_standard_spell_modifiers(rank_abilities: &mut Vec<RankAbility<'_>>) {
    rank_abilities.append(&mut vec![
        RankAbility {
            complexity: 0,
            name: "Spells",
            is_magical: true,
            rank: 1,
            description: "",
        },
        RankAbility {
            complexity: 0,
            name: "Spells",
            is_magical: true,
            rank: 2,
            description: "",
        },
        RankAbility {
            complexity: 0,
            name: "Spells",
            is_magical: true,
            rank: 3,
            description: "",
        },
        RankAbility {
            complexity: 0,
            name: "Spells",
            is_magical: true,
            rank: 4,
            description: "",
        },
        RankAbility {
            complexity: 0,
            name: "Spells",
            is_magical: true,
            rank: 5,
            description: "",
        },
        RankAbility {
            complexity: 0,
            name: "Spells",
            is_magical: true,
            rank: 6,
            description: "",
        },
        RankAbility {
            complexity: 0,
            name: "Spells",
            is_magical: true,
            rank: 7,
            description: "",
        },
    ]);
}

pub fn add_standard_maneuver_modifiers(rank_abilities: &mut Vec<RankAbility<'_>>) {
    rank_abilities.append(&mut vec![
        RankAbility {
            complexity: 0,
            name: "Maneuvers",
            is_magical: false,
            rank: 1,
            description: "",
        },
        RankAbility {
            complexity: 0,
            name: "Maneuvers",
            is_magical: false,
            rank: 3,
            description: "",
        },
        RankAbility {
            complexity: 0,
            name: "Maneuvers",
            is_magical: false,
            rank: 5,
            description: "",
        },
        RankAbility {
            complexity: 0,
            name: "Maneuvers",
            is_magical: false,
            rank: 7,
            description: "",
        },
    ]);
}

File: typescript/src/classes/definitions/standard_modifiers.ts
import { RankAbility } from '../types';

export function addStandardSpellModifiers(rankAbilities: RankAbility[]) {
  rankAbilities.push(
    { complexity: 0, name: 'Spells', isMagical: true, rank: 1, description: '' },
    { complexity: 0, name: 'Spells', isMagical: true, rank: 2, description: '' },
    { complexity: 0, name: 'Spells', isMagical: true, rank: 3, description: '' },
    { complexity: 0, name: 'Spells', isMagical: true, rank: 4, description: '' },
    { complexity: 0, name: 'Spells', isMagical: true, rank: 5, description: '' },
    { complexity: 0, name: 'Spells', isMagical: true, rank: 6, description: '' },
    { complexity: 0, name: 'Spells', isMagical: true, rank: 7, description: '' },
  );
}

export function addStandardManeuverModifiers(rankAbilities: RankAbility[]) {
  rankAbilities.push(
    { complexity: 0, name: 'Maneuvers', isMagical: false, rank: 1, description: '' },
    { complexity: 0, name: 'Maneuvers', isMagical: false, rank: 3, description: '' },
    { complexity: 0, name: 'Maneuvers', isMagical: false, rank: 5, description: '' },
    { complexity: 0, name: 'Maneuvers', isMagical: false, rank: 7, description: '' },
  );
}
